common4j/src/main/com/microsoft/identity/common/java/providers/microsoft/microsoftsts/MicrosoftStsAuthorizationRequest.java

```diff
@@ -20,47 +20,82 @@
 // LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
 // OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
 // THE SOFTWARE.
-package com.microsoft.identity.common.internal.providers.microsoft.microsoftsts;
-
-import android.net.Uri;
-import android.text.TextUtils;
-import android.util.Pair;
+package com.microsoft.identity.common.java.providers.microsoft.microsoftsts;
 
 import com.google.gson.annotations.Expose;
 import com.google.gson.annotations.SerializedName;
-import com.microsoft.identity.common.internal.providers.microsoft.MicrosoftAuthorizationRequest;
-import com.microsoft.identity.common.internal.providers.microsoft.azureactivedirectory.AzureActiveDirectorySlice;
-import com.microsoft.identity.common.java.util.ObjectMapper;
+import com.microsoft.identity.common.java.logging.Logger;
+import com.microsoft.identity.common.java.providers.microsoft.MicrosoftAuthorizationRequest;
+import com.microsoft.identity.common.java.providers.microsoft.azureactivedirectory.AzureActiveDirectorySlice;
+import com.microsoft.identity.common.java.util.StringUtil;
+
+import org.apache.http.client.utils.URIBuilder;
 
+import java.net.URI;
+import java.net.URISyntaxException;
 import java.util.HashMap;
 import java.util.Map;
 
+import lombok.Getter;
+import lombok.experimental.Accessors;
+
 public class MicrosoftStsAuthorizationRequest extends MicrosoftAuthorizationRequest<MicrosoftStsAuthorizationRequest> {
+    private static final Object TAG = MicrosoftStsAuthorizationRequest.class.getSimpleName();
+
     /**
      * Serial version id.
      */
     private static final long serialVersionUID = 6545759826515911472L;
 
-    private static final String AUTHORIZATION_ENDPOINT = "oAuth2/v2.0/authorize";
+    private static final String AUTHORIZATION_ENDPOINT = "/oAuth2/v2.0/authorize";
```

Comment:
No. This is constructed by hand. ApacheHTTP's URI builder doesn't support appending path.

That said, I can do something like this.

```Java
        final URIBuilder pathBuilder = new URIBuilder(AUTHORIZATION_ENDPOINT);
        final List<String> authEndpointSegment = pathBuilder.getPathSegments();

        final URIBuilder builder = new URIBuilder(this.getAuthority().toString());
        final List<String> pathSegments = builder.getPathSegments();
        pathSegments.addAll(authEndpointSegment);
        builder.setPathSegments(pathSegments);
        return builder.build().toString();
```

Reply:
I think this would be better, but we probably need some unit tests to verify the behavior.  I've seen double and triple slashes work, but they always lead to confusion when people examine logs... because they believe that must be the problem.

Reply:
Migrate the code above to UrlUtil and added test cases.